test_scripts/RC/OnRCStatus/commonOnRCStatus.lua

```diff
@@ -0,0 +1,250 @@
+      backTiltAngle = 10 ,
+      headSupportHorizontalPosition = 50 ,
+      headSupportVerticalPosition = 50 ,
+      massageEnabled = true ,
+      massageMode = {{
+        massageZone = "LUMBAR",
+        massageMode = "LOW"
+      }},
+      massageCushionFirmness = {{
+        cushion = "TOP_LUMBAR",
+        firmness = 10
+      }},
+      memory = {
+        id = 1,
+        label = "first",
+        action = "SAVE"
+      }
+    }
+  elseif "AUDIO" == pModuleType then
+    struct.moduleType = "AUDIO"
+    struct.audioControlData = {
+      source = "CD",
+      keepContext = false,
+      volume = 50,
+      equalizerSettings = {
+        channelId = 10,
+        channelName = "Channel 1",
+        channelSetting = 50 }
+    }
+  elseif "LIGHT" == pModuleType then
+    struct.moduleType = "LIGHT"
+    struct.lightControlData = {
+      lightState = {
+        id = "FRONT_LEFT_HIGH_BEAM",
+        status = "ON",
+        density = 0.2,
+        sRGBColor = "red"
+      }
+    }
+  elseif "HMI_SETTINGS" == pModuleType then
+    struct.moduleType = "HMI_SETTINGS"
+    struct.hmiSettingsControlData = {
+      displayMode = "DAY",
+      temperatureUnit = "CELSIUS",
+      distanceUnit = "KILOMETERS"
+    }
+  end
+  return struct
+end
+
+function commonOnRCStatus.postconditions()
+  commonRC.postconditions()
+end
+
+function commonOnRCStatus.SetModuleStatus(pFreeMod, pAllocatedMod, pModule)
+  local ModulesStatus = { }
+  table.insert(pAllocatedMod, pModule)
+  for key, value in pairs(pFreeMod) do
+    if pModule == value then
+      table.remove(pFreeMod, key)
+    end
+  end
+  ModulesStatus.freeModules = pFreeMod
+  ModulesStatus.allocatedModules = pAllocatedMod
+  return ModulesStatus
+end
+
+function commonOnRCStatus.SetModuleStatusByDeallocation(pFreeMod, pAllocatedMod, pModule)
+  local ModulesStatus = { }
+  table.insert(pFreeMod, pModule)
+  for key, value in pairs(pAllocatedMod) do
+    if pModule == value then
+      table.remove(pAllocatedMod, key)
+    end
+  end
+  ModulesStatus.freeModules = pFreeMod
+  ModulesStatus.allocatedModules = pAllocatedMod
+  return ModulesStatus
+end
+
+function commonOnRCStatus.unregisterApp(pAppId)
+  if not pAppId then pAppId = 1 end
+  commonRC.unregisterApp(pAppId, test)
+end
+
+function commonOnRCStatus.defineRAMode(pAllowed, pAccessMode)
+  commonRC.defineRAMode(pAllowed, pAccessMode, test)
+end
+
+function commonOnRCStatus.ConsentRejected(pModuleType, pRPC, pAppId)
+  if not pAppId then pAppId = 1 end
+  local mobSession = commonOnRCStatus.getMobileSession(pAppId)
+  local cid = mobSession:SendRPC(commonRC.getAppEventName(pRPC), commonRC.getAppRequestParams(pRPC, pModuleType))
+  local consentRPC = "GetInteriorVehicleDataConsent"
+  EXPECT_HMICALL(commonRC.getHMIEventName(consentRPC), commonRC.getHMIRequestParams(consentRPC, pModuleType, pAppId))
+  :Do(function(_, data)
+      test.hmiConnection:SendError(data.id, data.method, "REJECTED", " Driver rejects request ")
```

Comment:
@GetmanetsIrina Please notice in ```commonRC``` module there is a ```rpcRejectWithConsent``` function 
And the code is different.

Reply:
Locally changed.

Reply:
@dboltovskyi , rpcRejectWithConsent and current function have different logic. rpcRejectWithConsent has success consent, local function has rejected consent.